Added VehicleModel definitions.

r4sheed opened the pull request.

CLAassistant commented:
[![CLA assistant check](https://cla-assistant.io/pull/badge/not_signed)](https://cla-assistant.io/openmultiplayer/omp-stdlib?pullRequest=19) <br/>Thank you for your submission! We really appreciate it. Like many open source projects, we ask that you sign our [Contributor License Agreement](https://cla-assistant.io/openmultiplayer/omp-stdlib?pullRequest=19) before we can accept your contribution.<br/><sub>You have signed the CLA already but the status is still pending? Let us [recheck](https://cla-assistant.io/check/openmultiplayer/omp-stdlib?pullRequest=19) it.</sub>

Y-Less commented:
OK, I'll replicate my comments from discord here just for completeness:

1) While we like tags, vehicles weren't done originally because it seemed like an even more invasive change than weapons. Maybe that isn't the case, I don't now. As everyone who has read the documentation knows these extra checks can be disabled, but in terms of balance I don't know if this is too much. At least it is vehicle models, not vehicle ids.
2). Speaking of documentation, check the design document.
3) Don't add the tag to the short OPEN_MP_TAGS define, that's there for compatibility with old compilers that have a small limit on tag numbers.
4) Some of the spacing is off.
5) A few notes on the names, and missing variants:

* I haven't checked every name, but "artic" is short for "articulated" not "article" (I've just realised "article" is wrong in sscanf as well. Clearly we used a similar list).
* MR_WHOOPEE?
* RC_BANDIT and other `RC_` vehicles.
* TOPFUN_VAN
* PCJ_600, ZR_350, etc
* COMBONE_HARVESTER
* COMBINE*
* DUNE
* MAX is off by one

r4sheed commented:
> 1. While we like tags, vehicles weren't done originally because it seemed like an even more invasive change than weapons. Maybe that isn't the case, I don't now. As everyone who has read the documentation knows these extra checks can be disabled, but in terms of balance I don't know if this is too much. At least it is vehicle models, not vehicle ids.

I don't think vehicle models should be ignored. It should also be defined in the same way like weapons to achieve cleaner, more transparent code. It’s a pain to update game modes (and libraries) - but as you said it can be disabled so it’s optional. 

> 2). Speaking of documentation, check the design document.

Can you be more specific what I did wrong?

> 3. Don't add the tag to the short OPEN_MP_TAGS define, that's there for compatibility with old compilers that have a small limit on tag numbers.

I’ve removed it.

> 4. Some of the spacing is off.

I think it’s fixed now.

> 5. A few notes on the names, and missing variants:

The model definitions are based on the vehicle original entries from the .gxt files. 
For example `VEHICLE_MODEL_PCJ600` is `PCJ600=PCJ-600`, or `VEHICLE_MODEL_PETROTR` is `PETROTR=Petrol Truck`.
Also I’ve added some new variants that you mentioned above.

r4sheed commented:
> this pr is a horrible addition... it will make a lot of issues with original syntax (tag mismatch)... another reason for a headache!!!

Do you know what is horrible? Search for 'samp vehicle models' then find a `Sultan` then type `560` instead just typing `VEHICLE_SULTAN` in your code.

Zorono commented:
i am complaining about that change for modelid parameters' tag

shierru commented:
I would leave the list of vehicles, but I think `VEHICLE_MODEL:modelid` is a very terrible solution.

As for the list of vehicles and the ability to use a certain constant name instead of a model, which already has a certain meaning, I consider it a good practice.

But it shouldn’t be mandatory, otherwise it will simply become useless if you have to “rigidly” adapt the code to it.

Y-Less commented:
What we can do for now is use:

```pawn
#define VEHICLE_MODEL: _:
```

Instead of the define as-is.  This way we get the advantages of named constants and natives marked as requiring model IDs, but without actually having the tags yet.  And further down the line switching this out becomes almost trivial.

The PR does still need the `#define` versions of the names still though.

Trysha-rbrn commented:
> > this pr is a horrible addition... it will make a lot of issues with original syntax (tag mismatch)... another reason for a headache!!!
> 
> Do you know what is horrible? Search for 'samp vehicle models' then find a `Sultan` then type `560` instead just typing `VEHICLE_SULTAN` in your code. 

+1(kinda bump ig)
